.github/workflows/gradle-pr-build.yml

```diff
@@ -8,51 +8,34 @@ jobs:
   build:
     strategy:
       matrix:
-        os: [ubuntu-latest, windows-latest]
+        os: [ ubuntu-latest, windows-latest ]
     runs-on: ${{ matrix.os }}
     steps:
-    - uses: actions/checkout@v2
-    - name: Set up Java 8
-      uses: actions/setup-java@v2
-      with:
-        distribution: 'zulu'
-        java-version: 8
-    - name: Restore Gradle caches
-      uses: actions/cache@v2
-      with:
-        path: |
-          ~/.gradle/caches/modules-2
-          ~/.gradle/caches/files-2.1
-          ~/.gradle/caches/metadata-2.96
-          ~/.gradle/jdks
-        key: ${{ runner.os }}-gradle-cache-${{ hashFiles('**/*.gradle.kts') }}-${{ hashFiles('**/*.gradle') }}-${{ hashFiles('**/gradle.properties') }}
-        restore-keys: |
-          ${{ runner.os }}-gradle-cache-
-    - name: Restore Gradle wrapper
-      uses: actions/cache@v2
-      with:
-        path: |
-          ~/.gradle/wrapper
-        key: ${{ runner.os }}-gradle-wrapper-${{ hashFiles('**/gradlew') }}
-        restore-keys: |
-          ${{ runner.os }}-gradle-wrapper-
-    - name: Grant execute permission for gradlew
-      run: chmod +x gradlew
-    - name: Build with release Kotlin version
-      run: ./gradlew clean build ktlint
-    - name: Upload dependency verification report
-      uses: actions/upload-artifact@v2
-      if: failure()
-      with:
-        name: Build artifacts
-        path: '**/build/**/dependency-verification-report.html'
-        if-no-files-found: warn
-    - name: Upload test results
-      uses: actions/upload-artifact@v2
-      if: failure()
-      with:
-        name: Test artifacts
-        path: '**/tests/**/*.html'
-        if-no-files-found: warn
-    - name: Build with dev Kotlin version
-      run: ./gradlew -PkotlinDev clean build ktlint
+      - uses: actions/checkout@v3
+      - name: Set up Java 8
+        uses: actions/setup-java@v3
+        with:
+          distribution: 'zulu'
+          java-version: 8
+      - name: Build with release Kotlin version
+        uses: gradle/gradle-build-action@v2
```

Comment:
https://github.com/gradle/gradle-build-action

Ref https://github.com/detekt/detekt/pull/3723, it's really so fast to run jobs, and many other advantages.